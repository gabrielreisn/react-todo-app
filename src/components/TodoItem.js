/* @flow */

import * as React from 'react';
import Checkbox from 'material-ui/Checkbox';
import IconButton from 'material-ui/IconButton';
import Card from 'material-ui/Card';

import styled from 'styled-components';

import { cardStyle, checkBoxStyle, buttonStyle, iconStyle } from './styles';

const ItemWrapper = styled.div`
  display: flex;
  align-items: center;
`;

const CheckedText = styled.span`
  flex: 1 1 auto;
  text-decoration: ${props => (props.checked ? 'line-through' : '')};
`;

type Props = {
  text: string,
  onRemove: (element: number) => void,
};

type State = {
  checked: boolean,
};

class TodoItem extends React.Component<Props, State> {
  state = {
    checked: false,
  };

  toggleCheck = (): void => {
    this.setState(prevState => {
      return {
        checked: !prevState.checked,
      };
    });
  };

  render(): React.Node {
    const { checked } = this.state;
    const { onRemove, text } = this.props;

    return (
      <Card
        style={cardStyle.externalStyle}
        containerStyle={cardStyle.containerStyle}
      >
        <ItemWrapper>
          <Checkbox
            style={checkBoxStyle.checkBox}
            iconStyle={checkBoxStyle.selectedColor}
            checked={checked}
            onCheck={this.toggleCheck}
          />
          <CheckedText checked={checked}>{text}</CheckedText>
          <IconButton
            iconClassName="material-icons"
            style={buttonStyle}
            iconStyle={iconStyle}
            onClick={onRemove}
          >
            clear
          </IconButton>
        </ItemWrapper>
      </Card>
    );
  }
}

export default TodoItem;
