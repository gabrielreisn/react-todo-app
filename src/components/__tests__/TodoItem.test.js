import * as React from 'react';
import { shallow } from 'enzyme';
import TestRenderer from 'react-test-renderer';
import 'jest-styled-components';

import TodoItem from '../TodoItem';
import { CheckedText } from '../TodoItem';

describe('<TodoItem />', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<TodoItem />);
    expect(wrapper).toMatchSnapshot();
  });
  it('togglecheck should change boolean state', () => {
    const wrapper = shallow(<TodoItem />);
    expect(wrapper.state().checked).toEqual(false);
    wrapper.instance().toggleCheck();
    expect(wrapper.state().checked).toEqual(true);
  });
  it('changing the state should cross text', () => {
    const styledText = TestRenderer.create(
      <CheckedText checked={false} />
    ).toJSON();
    expect(styledText).toHaveStyleRule('text-decoration', 'none');

    //didn't found a way to update the proper component using react-test-renderer
    // neither a way to test styled components with enzyme

    const styledText2 = TestRenderer.create(<CheckedText checked />).toJSON();
    expect(styledText2).toHaveStyleRule('text-decoration', 'line-through');
  });
});
