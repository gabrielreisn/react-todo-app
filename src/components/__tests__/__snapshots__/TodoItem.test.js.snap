// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ForwardRef>
        <Checkbox
          checked={false}
          disabled={false}
          iconStyle={
            Object {
              "fill": "#ed7224",
            }
          }
          labelPosition="right"
          onCheck={[Function]}
          style={
            Object {
              "height": "24px",
              "marginRight": "20px",
              "width": "24px",
            }
          }
        />
        <ForwardRef
          checked={false}
        />
        <IconButton
          disableTouchRipple={false}
          disabled={false}
          iconClassName="material-icons"
          iconStyle={
            Object {
              "iconHoverColor": "#ed7224",
            }
          }
          style={
            Object {
              "margin": "auto",
              "padding": "0px",
            }
          }
          tooltipPosition="bottom-center"
          touch={false}
        >
          clear
        </IconButton>
      </ForwardRef>,
      "containerStyle": Object {
        "marginBottom": "15px",
      },
      "expandable": false,
      "expanded": null,
      "initiallyExpanded": false,
      "style": Object {
        "margin": "auto",
        "width": "75%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Checkbox
            checked={false}
            disabled={false}
            iconStyle={
              Object {
                "fill": "#ed7224",
              }
            }
            labelPosition="right"
            onCheck={[Function]}
            style={
              Object {
                "height": "24px",
                "marginRight": "20px",
                "width": "24px",
              }
            }
          />,
          <ForwardRef
            checked={false}
          />,
          <IconButton
            disableTouchRipple={false}
            disabled={false}
            iconClassName="material-icons"
            iconStyle={
              Object {
                "iconHoverColor": "#ed7224",
              }
            }
            style={
              Object {
                "margin": "auto",
                "padding": "0px",
              }
            }
            tooltipPosition="bottom-center"
            touch={false}
          >
            clear
          </IconButton>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": false,
            "disabled": false,
            "iconStyle": Object {
              "fill": "#ed7224",
            },
            "labelPosition": "right",
            "onCheck": [Function],
            "style": Object {
              "height": "24px",
              "marginRight": "20px",
              "width": "24px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checked": false,
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
  flex: 1 1 auto;
  text-decoration: ",
                [Function],
                ";
",
              ],
            },
            "displayName": "styled.span",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "span",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "clear",
            "disableTouchRipple": false,
            "disabled": false,
            "iconClassName": "material-icons",
            "iconStyle": Object {
              "iconHoverColor": "#ed7224",
            },
            "onClick": undefined,
            "style": Object {
              "margin": "auto",
              "padding": "0px",
            },
            "tooltipPosition": "bottom-center",
            "touch": false,
          },
          "ref": null,
          "rendered": "clear",
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  align-items: center;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ForwardRef>
          <Checkbox
            checked={false}
            disabled={false}
            iconStyle={
              Object {
                "fill": "#ed7224",
              }
            }
            labelPosition="right"
            onCheck={[Function]}
            style={
              Object {
                "height": "24px",
                "marginRight": "20px",
                "width": "24px",
              }
            }
          />
          <ForwardRef
            checked={false}
          />
          <IconButton
            disableTouchRipple={false}
            disabled={false}
            iconClassName="material-icons"
            iconStyle={
              Object {
                "iconHoverColor": "#ed7224",
              }
            }
            style={
              Object {
                "margin": "auto",
                "padding": "0px",
              }
            }
            tooltipPosition="bottom-center"
            touch={false}
          >
            clear
          </IconButton>
        </ForwardRef>,
        "containerStyle": Object {
          "marginBottom": "15px",
        },
        "expandable": false,
        "expanded": null,
        "initiallyExpanded": false,
        "style": Object {
          "margin": "auto",
          "width": "75%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Checkbox
              checked={false}
              disabled={false}
              iconStyle={
                Object {
                  "fill": "#ed7224",
                }
              }
              labelPosition="right"
              onCheck={[Function]}
              style={
                Object {
                  "height": "24px",
                  "marginRight": "20px",
                  "width": "24px",
                }
              }
            />,
            <ForwardRef
              checked={false}
            />,
            <IconButton
              disableTouchRipple={false}
              disabled={false}
              iconClassName="material-icons"
              iconStyle={
                Object {
                  "iconHoverColor": "#ed7224",
                }
              }
              style={
                Object {
                  "margin": "auto",
                  "padding": "0px",
                }
              }
              tooltipPosition="bottom-center"
              touch={false}
            >
              clear
            </IconButton>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "disabled": false,
              "iconStyle": Object {
                "fill": "#ed7224",
              },
              "labelPosition": "right",
              "onCheck": [Function],
              "style": Object {
                "height": "24px",
                "marginRight": "20px",
                "width": "24px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "checked": false,
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": false,
                "rules": Array [
                  "
  flex: 1 1 auto;
  text-decoration: ",
                  [Function],
                  ";
",
                ],
              },
              "displayName": "styled.span",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "span",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "clear",
              "disableTouchRipple": false,
              "disabled": false,
              "iconClassName": "material-icons",
              "iconStyle": Object {
                "iconHoverColor": "#ed7224",
              },
              "onClick": undefined,
              "style": Object {
                "margin": "auto",
                "padding": "0px",
              },
              "tooltipPosition": "bottom-center",
              "touch": false,
            },
            "ref": null,
            "rendered": "clear",
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  align-items: center;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
